drivers/hmis/app/models/hmis/hud/processors/base.rb

```diff
@@ -290,6 +301,18 @@ def normalize_single_custom_field_value(cded, value)
       else
         raise "unexpected value \"#{value}\""
       end
+    when 'file'
+      case value
+      when String
+        value
+      when Object
+        # If the value is an Object, it should be a File record in our db
+        raise "unexpected value \"#{value}\"" unless value.key?('fileBlobId') && value.key?('id')
+
+        Hmis::File.find(value['id'])
```

Comment:
I think it's a little confusing for this lookup to be in the normalization function instead of in the `process_blob_id_to_file` method, can we move it down?

Reply:
Thanks, yeah I agree it is confusing. After reading your feedback and taking another look, I think it would be best not to pass around the whole file object. I'm almost there with a tweaked approach - coming soon ...

Reply:
Ok, I think this ready for re-review. 
This won't work with regular blob ID, only a signed blob ID; so I think it would need to go out in the next release after https://github.com/greenriver/hmis-frontend/pull/987. But I'd like to think this through a little more together.

Reply:
Thanks @martha this looks good! Yes as discussed let's re-target this to 145